noir_stdlib/src/ops/arith.nr

```diff
@@ -346,3 +348,227 @@ impl Neg for i64 {
+    fn wrapping_add(self: u8, y: u8) -> u8 {
+        wrapping_add_hlp(self, y)
+    }
+}
+impl WrappingSub for u8 {
+
+    fn wrapping_sub(self: u8, y: u8) -> u8 {
+        wrapping_sub_hlp(self, y)
+    }
+}
+impl WrappingMul for bool {
+    fn wrapping_mul(self: bool, y: bool) -> bool {
+        wrapping_mul_hlp(self, y)
+    }
+}
+impl WrappingAdd for bool {
+    fn wrapping_add(self: bool, y: bool) -> bool {
+        wrapping_add_hlp(self, y)
+    }
+}
+impl WrappingSub for bool {
+    fn wrapping_sub(self: bool, y: bool) -> bool {
+        wrapping_sub_hlp(self, y)
+    }
+}
+impl WrappingMul for u128 {
+    fn wrapping_mul(self: u128, y: u128) -> u128 {
+        wrapping_mul128_hlp(self as Field, y as Field) as u128
+    }
+}
+impl WrappingAdd for u128 {
+    fn wrapping_add(self: u128, y: u128) -> u128 {
+        wrapping_add_hlp(self, y)
+    }
+}
+impl WrappingSub for u128 {
+    fn wrapping_sub(self: u128, y: u128) -> u128 {
+        wrapping_sub_hlp(self, y)
+    }
+}
+
+impl WrappingMul for i64 {
+    fn wrapping_mul(self: i64, y: i64) -> i64 {
+        wrapping_mul_hlp(self, y)
+    }
+}
+impl WrappingAdd for i64 {
+    fn wrapping_add(self: i64, y: i64) -> i64 {
+        wrapping_add_hlp(self, y)
+    }
+}
+impl WrappingSub for i64 {
+    fn wrapping_sub(self: i64, y: i64) -> i64 {
+        wrapping_sub_hlp(self, y)
+    }
+}
+
+impl WrappingMul for i32 {
+    fn wrapping_mul(self: i32, y: i32) -> i32 {
+        wrapping_mul_hlp(self, y)
+    }
+}
+impl WrappingAdd for i32 {
+    fn wrapping_add(self: i32, y: i32) -> i32 {
+        wrapping_add_hlp(self, y)
+    }
+}
+impl WrappingSub for i32 {
+    fn wrapping_sub(self: i32, y: i32) -> i32 {
+        wrapping_sub_hlp(self, y)
+    }
+}
+impl WrappingMul for i16 {
+    fn wrapping_mul(self: i16, y: i16) -> i16 {
+        wrapping_mul_hlp(self, y)
+    }
+}
+impl WrappingAdd for i16 {
+    fn wrapping_add(self: i16, y: i16) -> i16 {
+        wrapping_add_hlp(self, y)
+    }
+}
+impl WrappingSub for i16 {
+    fn wrapping_sub(self: i16, y: i16) -> i16 {
+        wrapping_sub_hlp(self, y)
+    }
+}
+impl WrappingMul for i8 {
+    fn wrapping_mul(self: i8, y: i8) -> i8 {
+        wrapping_mul_hlp(self, y)
+    }
+}
+impl WrappingAdd for i8 {
+    fn wrapping_add(self: i8, y: i8) -> i8 {
+        wrapping_add_hlp(self, y)
+    }
+}
+impl WrappingSub for i8 {
+    fn wrapping_sub(self: i8, y: i8) -> i8 {
+        wrapping_sub_hlp(self, y)
+    }
+}
+
+fn wrapping_add_hlp<T>(x: T, y: T) -> T
+where
+    T: AsPrimitive<Field>,
+    Field: AsPrimitive<T>,
+{
+    AsPrimitive::as_(x.as_() + y.as_())
+}
+
+fn wrapping_sub_hlp<T>(x: T, y: T) -> T
+where
+    T: AsPrimitive<Field>,
+    Field: AsPrimitive<T>,
+{
+    //340282366920938463463374607431768211456 is 2^128, it is used to avoid underflow
+    AsPrimitive::as_(x.as_() + 340282366920938463463374607431768211456 - y.as_())
+}
+
+fn wrapping_mul_hlp<T>(x: T, y: T) -> T
+where
+    T: AsPrimitive<Field>,
+    Field: AsPrimitive<T>,
```

Comment:
oof, that's a doozy. We shouldn't allow this in the typesystem and we should have a `FromPrimitive` trait imo

Reply:
Ah, I must have never run into a situation where I needed this before.